### PR TITLE
test: merge blocks when prepare

### PR DESCRIPTION
**Pull request by breezewish**
Signed-off-by: Wish <breezewish@outlook.com>

### What problem does this PR solve?

Issue Number: ref #5237

Problem Summary:

### What is changed and how it works?

Always produce a single sorted block, instead of a list of blocks, when we need to fill rows in tests. In this way, we can produce a correct DMFile from these rows.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/tests/gtest_segment_test_basic.cpp [hunk 1]
```diff
@@ -301,7 +301,34 @@ Block SegmentTestBasic::prepareWriteBlock(Int64 start_key, Int64 end_key, bool i
         is_deleted);
 }
 
-std::vector<Block> SegmentTestBasic::prepareWriteBlocksInSegmentRange(PageId segment_id, UInt64 total_write_rows, std::optional<Int64> write_start_key, bool is_deleted)
+Block mergeBlocks(std::vector<Block> && blocks)
+{
+    auto accumulated_block = std::move(blocks[0]);
+
+    for (size_t block_idx = 1; block_idx < blocks.size(); block_idx++)
```
Comment (Lloyd-Pottiger) [inline dbms/src/Storages/DeltaMerge/tests/gtest_segment_test_basic.cpp:308]:
```suggestion
    for (size_t block_idx = 1; block_idx < blocks.size(); ++block_idx)
```

File: dbms/src/Storages/DeltaMerge/tests/gtest_sst_files_stream.cpp [hunk 2]
```diff
@@ -430,4 +430,4 @@ CATCH
 
 } // namespace tests
 } // namespace DM
-} // namespace DB
\ No newline at end of file
+} // namespace DB
```
Comment (Lloyd-Pottiger) [inline dbms/src/Storages/DeltaMerge/tests/gtest_sst_files_stream.cpp]: merge the namespace into "DB::DM::tests"?

**Review by Lloyd-Pottiger — APPROVED** [review]
nice work!

**Review by JaySon-Huang — APPROVED** [review]

**Pull request merged**